src/smt/theory_str.cpp

```diff
@@ -1880,6 +1880,16 @@ namespace smt {
             SASSERT(axiom1);
             assert_axiom(axiom1);
         }
+
+        // axiom 2: The only (str.from-int N) that starts with a "0" is "0".
+        expr_ref zero(mk_string("0"), m);
+        // let (the result starts with a "0") be p
+        expr_ref p(u.str.mk_prefix(zero, ex), m);
+        // let (the result is "0") be q  
+        expr_ref q(ctx.mk_eq_atom(ex, zero), m);
+        // encoding: the result does NOT start with a "0" (~p) xor the result is "0" (q)
+        // ~p xor q == (~p or q) and (p or ~q)
+        assert_axiom(m.mk_and(m.mk_or(mk_not(m, p), q), m.mk_or(p, mk_not(m, q))));
```

Comment:
Is there a reason why the function form of `mk_not` was used and not the method on `ast_manager` (i.e. `m.mk_not(p)`)?

Reply:
To be consistent with the existing code (e.g. line 1877 right above).